liblttng-ust-java-agent/java/lttng-ust-agent-jul/org/lttng/ust/agent/jul/LttngLogHandler.java

```diff
@@ -134,13 +156,17 @@ public void publish(LogRecord record) {
 		 */
 		LttngJulApi.tracepointWithContext(formattedMessage,
 				record.getLoggerName(),
-				record.getSourceClassName(),
-				record.getSourceMethodName(),
+				logSource ? record.getSourceClassName() : EMPTY,
+				logSource ? record.getSourceMethodName() : EMPTY,
 				record.getMillis(),
 				record.getLevel().intValue(),
 				record.getThreadID(),
 				contextInfo.getEntriesArray(),
 				contextInfo.getStringsArray());
 	}
 
+	public void setLogSource(boolean doLog) {
```

Comment:
Those kinds of classes usually have setters for individual properties. So I'd be tempted to keep the setters as it can be set at different moments of the trace, if anyone wishes it. Especially those using this handler directly.